packages/typescript-estree/src/node-utils.ts

```diff
@@ -450,11 +450,11 @@ export function isOptional(node: {
  * @param token the ts.Token
  * @returns the token type
  */
-// ts.Node types are ugly
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export function getTokenType(token: any): AST_TOKEN_TYPES {
+export function getTokenType(
+  token: ts.Identifier | ts.Token<ts.SyntaxKind>,
+): AST_TOKEN_TYPES {
   // Need two checks for keywords since some are also identifiers
-  if (token.originalKeywordKind) {
+  if ('originalKeywordKind' in token) {
```

Comment:
nit this works differently than the original check.
Just highlighting to make sure that the logic change is intended.

`in` will only ensure the prop exists, not that it's non-nullish.
So this change will mean a nullish value will fall through to the default switch case instead of skipping this branch

eg

```ts
const x = {originalKeywordKind: null}

if ('originalKeywordKind' in x) { console.log('is in') }
if (x.originalKeywordKind) { console.log('is truthy') }
```

will only output:
```
is in
```

Reply:
in this case this field is available only for Identifier and has to be undefined or present. I will update it to
```suggestion
  if ('originalKeywordKind' in token && 'originalKeywordKind' in token) {
```